### PR TITLE
fix(sf:number): fix show undefined or null when use prefix or unit an…

### PR DESCRIPTION
**Pull request by danranVm**
…d value is undefined or null

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ng-alain/delon/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: [ng-alain/ng-alain/#968](https://github.com/ng-alain/ng-alain/issues/968)

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/form/src/widgets/number/number.widget.ts [hunk 1]
```diff
@@ -27,11 +27,11 @@ export class NumberWidget extends ControlWidget implements OnInit {
       this.step = Math.trunc(this.step);
     }
     if (ui.prefix != null) {
-      ui.formatter = value => `${ui.prefix} ${value}`;
+      ui.formatter = value => (value === undefined || value === null) ? '' : `${ui.prefix} ${value}`;
```
Comment (cipchk) [inline packages/form/src/widgets/number/number.widget.ts]:
```suggestion
      ui.formatter = value => value == null ? '' : `${ui.prefix} ${value}`;
```

File: packages/form/src/widgets/number/number.widget.ts [hunk 2]
```diff
@@ -27,11 +27,11 @@ export class NumberWidget extends ControlWidget implements OnInit {
...
       ui.parser = value => value.replace(`${ui.prefix} `, '');
     }
     if (ui.unit != null) {
-      ui.formatter = value => `${value} ${ui.unit}`;
+      ui.formatter = value => (value === undefined || value === null) ? '' : `${value} ${ui.unit}`;
```
Comment (cipchk) [inline packages/form/src/widgets/number/number.widget.ts]:
```suggestion
      ui.formatter = value => value == null ? '' : `${value} ${ui.unit}`;
```

File: packages/form/src/widgets/number/number.widget.spec.ts [hunk 3]
```diff
@@ -101,19 +101,27 @@ describe('form: widget: number', () => {
   describe('[ui]', () => {
     it('#prefix', fakeAsync(() => {
       const s: SFSchema = {
-        properties: { a: { type: 'number', default: 1, ui: { prefix: 'a' } } },
+        properties: { a: { type: 'number', default: undefined, ui: { prefix: 'a' } } },
```
Comment (cipchk) [inline packages/form/src/widgets/number/number.widget.spec.ts]: 建议不要修改现有测试用例规则，请提供新的单元测试来满足当前修改的状态。

File: packages/form/src/widgets/number/number.widget.ts (hunk 2 above)
Comment (danranVm) [reply packages/form/src/widgets/number/number.widget.ts]: done

File: packages/form/src/widgets/number/number.widget.spec.ts (hunk 3 above)
Comment (danranVm) [reply packages/form/src/widgets/number/number.widget.spec.ts]: 小改了一下，不影响原测试用例。

**Comment by cipchk** [comment]
@danranVm If throw `screenfull` error, pls try merge master.

**Pull request merged**